pkg/vsphere/disk/disk_manager.go

```diff
@@ -469,7 +469,7 @@ func (m *Manager) detach(op trace.Operation, disk *types.VirtualDisk) error {
 		return t, er
 	})
 
-	if err != nil {
+	if err == nil {
```

Comment:
Can we add some log here to see whether the maxAttached is true or not before `if`? We don't even know whether this would be an error or not